protected/humhub/modules/content/widgets/stream/StreamEntryWidget.php

```diff
@@ -102,15 +102,31 @@ private static function renderLegacyWallEntry(ContentActiveRecord $record, $opti
             $options = [];
         }
 
+        // Ensure that $record->getWallEntryWidget() returns a valid object
+        $wallEntryWidget = $record->getWallEntryWidget();
+        if (!$wallEntryWidget || !$wallEntryWidget->renderOptions) {
+            // Handle the case where $wallEntryWidget or renderOptions is null
+            return;
+        }
+
+        // Ensure that $wallEntryWidget->renderOptions is properly initialized before accessing its properties
+        $jsWidget = null;
         if (isset($options['jsWidget'])) {
             $jsWidget = $options['jsWidget'];
             unset($options['jsWidget']);
         } else {
-            $jsWidget = $record->getWallEntryWidget()->jsWidget;
+            $jsWidget = $wallEntryWidget->renderOptions->jsWidget;
+        }
+
+        // Check if $record has getWallOut method
+        if (method_exists($record, 'getWallOut')) {
+            $content = $record->getWallOut($options);
+        } else {
+            $content = '';
         }
 
         $params = [
-            'content' => $record->getWallOut($options),
+            'content' => $content,
```

Comment:
Instead of the `method_exists()` we prefer to use a code like:

```php
'content' => $record instanceof ContentActiveRecord ? $record->getWallOut($options) : ''
```

Reply:
I can confirm this is a better option!